build-system/tasks/create-module-compatible-es5-bundle.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const $$ = require('gulp-load-plugins')();
+const gulp = $$.help(require('gulp'));
+const gulpRename = require('gulp-rename');
+const gulpReplace = require('gulp-replace');
+/* Copy source to source-nomodule.js and
+ * make it compatible with `<script type=module`.
+ * The regex replacement is ClosureCompiler specific, and will not be required
+ * for an ES6 bundle.
+ *
+ * Finds and replaces regex changing `this` -> `self` in the snippet
+ * inserted by closure compiler from
+ * https://github.com/google/closure-compiler/blob/36f332788d54803c3c1afe06a9d84bf4b9f4945b/src/com/google/javascript/jscomp/js/util/global.js#L44
+ */
+exports.createModuleCompatibleES5Bundle = function(src) {
+  return gulp.src('dist/' + src)
+      .pipe(gulpRename(src.substring(0, src.indexOf('.js')) + '-module.js'))
+      .pipe(gulpReplace(/global\?global:[a-z]*}\(this\)/, function(string) {
```

Comment:
Can you give a sample of what will be replaced?

Reply:
responded on chat